extensions/azurePublish/src/services/blobStorageService.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { StorageManagementClient } from '@azure/arm-storage';
+import { TokenCredentials } from '@azure/ms-rest-js';
+
+import { throwNotImplementedError } from './throwNotImplementedError';
+
+const createBlobStorageService = (token: string, subscriptionId: string) => {
+  const tokenCredentials = new TokenCredentials(token);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const storageManagementClient = new StorageManagementClient(tokenCredentials, subscriptionId);
+
+  const checkNameAvailability = async () => {
```

Comment:
Consider having a checkNameAvailability for any Azure resource (if they have to be unique across the entire Azure system) or  having this method on each service (if they have to be unique for that service type).  I suspect that Azure requires all resource names to be universally unique.